factory-levels/control.lua

```diff
@@ -81,17 +81,58 @@ function determine_level(finished_products_count)
 	return should_have_level
 end
 
+function determine_machine(entity)
+	if entity == nil or not entity.valid or (entity.type ~= "assembling-machine" and entity.type ~= "furnace") then
+		return nil
+	end
+	for _, machine in pairs(machines) do
+		if entity.name == machine.name or string_starts_with(entity.name, machine.level_name) then
+			return machine
+		end
+	end
+	return nil
+end
+
 function upgrade_factory(surface, targetname, sourceentity)
 	local count = sourceentity.products_finished
 	local box = sourceentity.bounding_box
+	local item_requests = nil
+	local recipe = nil
+	
+	local existing_requests = surface.find_entity("item-request-proxy", sourceentity.position)
```

Comment:
what is item-request-proxy? and what are existing requests? are these logistic robot requests?

Reply:
https://lua-api.factorio.com/latest/LuaEntity.html#LuaEntity.item_requests